### DISCUSSION TITLE
Multiple times loading checkpoint during indexing

### DISCUSSION THREAD

**Discussion by marcbarb** [stanford-futuredata/ColBERT#214] [Q&A]
It looks to me like the checkpoint is being loaded multiple times (an expensive operation) when indexing data.  Calls to ColBERTConfig.load_from_checkpoint are made on line 23 of colbert/indexer and line 23 of colbert/modeling/base_colbert.  Is there a reason for this.  Am I missing something?

Marc

**Comment by okhat** [comment]
Is it causing an issue?

**Comment by marcbarb** [comment]
Sorry, no it isn't.  And now when I run things in debug mode I see that it doesn't take that long.  I was just wondering why it was done twice.  No big deal.
